### PR TITLE
Use ffizz-header

### PR DESCRIPTION
**Pull request by djmitche**
#### Description

This switches the generation of `taskchampion.h` to use `ffizz-header` (which I wrote). This drops use of `cbindgen`, which did not allow sufficient control over the generated header.

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
       - name: Setup mdBook
         uses: peaceiris/actions-mdbook@v1
         with:
-          # if this changes, change it in cli/Cargo.toml and .github/workflows/publish-docs.yml as well
+          # if this changes, change it in .github/workflows/publish-docs.yml as well
```
Comment (ryneeverett) [inline .github/workflows/publish-docs.yml]: This comment created an infinite loop in my brain. Should the referenced file be changed to `
.github/workflows/checks.yml`?

Comment (djmitche) [reply .github/workflows/publish-docs.yml]: Thanks - fixed!

**Comment by djmitche** [comment]
This also fixes #2870, by adding a `taskwarrior_lib::generate_header` function that returns the header. Embedders can use this in their build process.

**Comment by djmitche** [comment]
I'm merging this on the assumption that (a) someone has looked at it (thanks @ryneeverett!) and (b) it has no user-visible impact.

**Pull request merged**